File: Algebra_Simultaneous2Unknown.cpp
// Algebra.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "Simultaneous2Unknown.h"

#include <iostream>
#include <vector>
#include <exception>

using namespace std;

int main() {

	cout << "\n    Welcome to our demonstration sequels";
	cout << "\n    Hope you enjoy (and follow) the lessons.\n\n";

	vector<unsigned> numerators;
	vector<unsigned> denominators;

	try {
		/*
		* Simultaneous Equations with 2 unknowns
		*/
		char * infinity = "infinity";
		char operators2D[]{ '+', '+' };
		double x_coeff2D[]{ 2, 1 };
		double y_coeff2D[]{ -1, 1 };
		double equals2D[]{ 1, 2 };

		if (y_coeff2D[0] < 0) {
			operators2D[0] = '-';
		}
		if (y_coeff2D[1] < 0) {
			operators2D[1] = '-';
		}

		cout << "\n    Solving simultaneously the equations:\n";
		//Print as an equation
		printf("%40.2fx %c %.2fy = %.2f\n", x_coeff2D[0], operators2D[0], abs(y_coeff2D[0]), equals2D[0]);
		printf("%40.2fx %c %.2fy = %.2f\n", x_coeff2D[1], operators2D[1], abs(y_coeff2D[1]), equals2D[1]);
		printf("\n%30s\n%40s", "Yields:", "(x, y)  =  ");

		try {
			Simultaneous2Unknown sim2unk(x_coeff2D, y_coeff2D, equals2D);
			sim2unk.solveSimultaneous();

			printf("(%.4f, %.4f)\n", sim2unk.x_variable, sim2unk.y_variable);

		}
		catch (...) {
			printf("(%s, %s)\n", infinity, infinity);
		}

		cout << "\n\n";

	} catch (exception& e) {
		cout << "\n" << e.what() << "\n";
	}

    return 0;
}

File: Algebra_Simultaneous2Unknown.vb
Module Algebra_2Unknown

    Sub Main()
        ''
        ' Simultaneous Equations With 2 unknowns
        ''
        Dim operators = {"+", "+"}
        Dim x_coefficients = {8.0, 2.0}
        Dim y_coefficients = {-5.0, 17.0}
        Dim equal = {89.0, 150.0}

        Dim equations As New Dictionary(Of String, Double())
        equations.Add("x", x_coefficients)
        equations.Add("y", y_coefficients)
        equations.Add("eq", equal)

        If y_coefficients(0) < 0 Then
            operators(0) = "-"
        End If
        If y_coefficients(1) < 0 Then
            operators(1) = "-"
        End If

        Console.WriteLine(vbCrLf, "    Solving simultaneously the equations:", vbCrLf)
        'Print As an equation
        Console.WriteLine(String.Format("{0,20}x  {1}  {2}y  =  {3}",
              x_coefficients(0), operators(0), Math.Abs(y_coefficients(0)), equal(0)
              )
        )
        Console.WriteLine(String.Format("{0,20}x  {1}  {2}y  =  {3}",
            x_coefficients(1), operators(1), Math.Abs(y_coefficients(1)), equal(1)
            )
        )
        Console.WriteLine(Environment.NewLine)
        Console.Write(String.Format("{0,15} {1}{2,20}", "Yields:", vbCrLf, "(x, y)  =  "))

        Try
            Dim sim2unk As New Simultaneous2Unknown
            sim2unk._init_(equations)
            Dim solution = sim2unk.solveSimultaneous()

            Console.Write(String.Format("({0:0.0000}, {1:0.0000})", solution(0), solution(1)))

        Catch ex As Exception
            Console.Write("(infinity,  infinity)")
        End Try

    End Sub

End Module
